File: src/main/kotlin/services/DashboardService.java
package services;

import models.JPA.Employee;
import models.JPA.EmployeeDto;
import models.JPA.Title;
import exceptions.InvalidInput;
import com.dashboard.repositories.EmployeeRepository;
import interfaces.serviceInterfaces.DashboardServiceInterface;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class DashboardService implements DashboardServiceInterface {

    @Autowired
    private EmployeeRepository employeeRepository;

    @Override
    public void createAccount(EmployeeDto employeeDto) throws InvalidInput {

        if(!employeeDto.getPassword1().equalsIgnoreCase(employeeDto.getPassword2())){
            throw new InvalidInput("Passwords do not match");
        }

        Employee employee = new Employee(employeeDto.getUsername(), employeeDto.getFirstName(), employeeDto.getLastName(), employeeDto.getPassword1(), employeeDto.getSalary(), Title.valueOf(employeeDto.getPosition()));
        employeeRepository.save(employee);
        log.debug("New employee is saved {}", employee);
    }

}

File: src/main/kotlin/services/DashboardService.kt
package services

import dashboard.models.jpa.Employee
import exceptions.InvalidInput
import interfaces.serviceInterfaces.DashboardServiceInterface
import models.jpa.EmployeeDto
import models.jpa.Title
import org.springframework.stereotype.Service
import repositories.EmployeeRepository

@Service
class DashboardService(val employeeRepository: EmployeeRepository) : DashboardServiceInterface {

    @Throws(InvalidInput::class)
    override fun createAccount(employeeDto: EmployeeDto) {
        if (!employeeDto!!.password1.equals(employeeDto.password2, ignoreCase = true)) {
            throw InvalidInput("Passwords do not match")
        }
        val employee = Employee(
            username = employeeDto.username,
            firstname = employeeDto.firstName,
            lastname = employeeDto.lastName,
            password = employeeDto.password1,
            salary = employeeDto.salary,
            title = Title.valueOf(employeeDto.position)
        )
        employeeRepository.save(employee)
//        DashboardService.log.debug("New employee is saved {}", employee)
    }
}